overlord/servicestate/quota_control.go

```diff
@@ -81,21 +81,72 @@ func quotaGroupsAvailable(st *state.State) error {
 	return nil
 }
 
+// MockQuotaInState mocks a given quota group in the state without ensuring any
+// of the on-disk state exists. It is meant to be consumed only by tests.
+func MockQuotaInState(st *state.State, name string, parentName string, snaps []string, memoryLimit quantity.Size) error {
```

Comment:
I'm really not a fan of this living here, we might have to introduce both an internal and a servicestatetest package to avoid this though.  devicestate does that for similar reasons

Reply:
hmm, I'm not sure how to move this function to another package without exporting things we probably don't want to export from this package or without duplicating the implementation, for example this function uses `quotaCreateImpl` which uses `patchQuotas`, so either I could export one or both of those functions or I could duplicate the implementation of one or both of those functions.

I guess the other thing I could do is some sort of hook situation, but that gets weird to do since servicestatetest needs to import servicestate, so any hooks that are exported in servicestatetest can't be assigned to from servicestate.

Reply:
that's why I mentioned that you would probably need to introduce a servicestate/internal package. I can try to explore this a bit more concretely

Reply:
I did the prerequisites steps of what I described/had in mind in https://github.com/snapcore/snapd/pull/10429

Reply:
this helper is now in servicestatetest, but I introduced it as part of a smaller PR: https://github.com/snapcore/snapd/pull/10434 so please review that one first